torch/custom_class_detail.h

```diff
@@ -51,6 +37,51 @@ struct pickle_factory<Get, Set, RetState(Self), NewInstance(ArgState)> {
   Set s;
 };
 
+template <typename Method>
+struct WrapMethod;
+
+template <typename R, typename CurrClass, typename... Args>
+struct WrapMethod<R (CurrClass::*)(Args...)> {
+  WrapMethod(R (CurrClass::*m)(Args...)) : m(std::move(m)) {}
+
+  R operator()(c10::intrusive_ptr<CurrClass> cur, Args... args) {
+    return c10::guts::invoke(m, *cur, args...);
+  }
+
+  R (CurrClass::*m)(Args...);
+};
+
+template <typename R, typename CurrClass, typename... Args>
+struct WrapMethod<R (CurrClass::*)(Args...) const> {
+  WrapMethod(R (CurrClass::*m)(Args...) const) : m(std::move(m)) {}
+
+  R operator()(c10::intrusive_ptr<CurrClass> cur, Args... args) {
+    return c10::guts::invoke(m, *cur, args...);
+  }
+
+  R (CurrClass::*m)(Args...) const;
+};
```

Comment:
It would be nice if these two specialization could be the same thing. (they're not right now because of `const`)

Reply:
It's probably possible to unify them into one but I'm not sure if it's worth the effort. Ping me offline if you want to try.